future-apps/staking/contracts/Staking.sol

```diff
@@ -131,15 +146,30 @@ contract Staking is ERCStaking, AragonApp {
     authP(STAKE_ROLE, arr(amount))
     authP(LOCK_ROLE, arr(amount, uint256(lockUnit), uint256(lockEnds)))
     public
+    returns(uint256 lockId)
   {
     stake(amount, stakeData);
-    lock(amount, lockUnit, lockEnds, unlocker, metadata, lockData);
+    return lock(amount, lockUnit, lockEnds, unlocker, metadata, lockData);
   }
 
-  function unlockAll(address acct) external {
+  function unlockAllOrNone(address acct) external {
     while (accounts[acct].locks.length > 0) {
-      if (canUnlock(acct, 0)) {
-        unlock(acct, 0);
+      uint256 lastAccountLock = accounts[acct].locks.length - 1;
```

Comment:
I like the idea of this function, but we shouldn't re-implement unlock here. Why not just iterate over all as in `unlockAll`?

Reply:
Well, yes, I'm "re-implementing", but it's only 2 things: remove last array element and emitting event.
I don't want to call `unlock` because then, in case of success, I would be repeating the `canUnlock` and `lastAccountLock` things. Yes, not big deal, but I find it ugly.
With respect `unlockAll`, I'm iterating over all of them, but with a `while` instead of a `for`, which I find nicer. In `unlockAll` I can not do the same because in the end the array may not be empty.